nifi-provenance-reporting-tasks/src/main/java/io/egm/nifi/reporting/ElasticsearchProvenanceReporter.java

```diff
@@ -87,46 +88,55 @@ public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
-        preparedEvent.put("process_group_id", event.get("process_group_id"));
-        preparedEvent.put("event_type", event.get("event_type"));
-        preparedEvent.put("status", event.get("status"));
-        preparedEvent.put("download_input_content_uri", event.get("download_input_content_uri"));
-        preparedEvent.put("download_output_content_uri", event.get("download_output_content_uri"));
-        preparedEvent.put("view_input_content_uri", event.get("view_input_content_uri"));
-        preparedEvent.put("view_output_content_uri", event.get("view_output_content_uri"));
-        preparedEvent.put("updated_attributes", objectMapper.writeValueAsString(event.get("updated_attributes")));
-        preparedEvent.put("previous_attributes", objectMapper.writeValueAsString(event.get("previous_attributes")));
-        if (event.containsKey("details"))
-            preparedEvent.put("details", event.get("details"));
-
-        final IndexRequest<Map<String, Object>> indexRequest = new
-                IndexRequest.Builder<Map<String, Object>>()
-                .index(elasticsearchIndex)
-                .id(id)
-                .document(preparedEvent)
-                .build();
-        try {
-            client.index(indexRequest);
-        } catch (ElasticsearchException ex) {
-            getLogger().error("Error while indexing event {}", id, ex);
-        }
+        events.forEach(event -> {
+            final String id = Long.toString((Long) event.get("event_id"));
+
+            if (!event.containsKey("process_group_name") || !event.containsKey("component_name")) {
+                getLogger().warn("Provenance event has no process group or processor, ignoring");
+                return;
+            }
+
+            Map<String, Object> preparedEvent = new HashMap<>();
+            preparedEvent.put("event_time_millis", event.get("event_time"));
+            preparedEvent.put("event_time_iso_utc", event.get("event_time_iso_utc"));
+            preparedEvent.put("component_type", event.get("component_type"));
+            preparedEvent.put("component_url", event.get("component_url"));
+            preparedEvent.put("component_name", event.get("component_name"));
+            preparedEvent.put("process_group_name", event.get("process_group_name"));
+            preparedEvent.put("process_group_id", event.get("process_group_id"));
+            preparedEvent.put("event_type", event.get("event_type"));
+            preparedEvent.put("status", event.get("status"));
+            preparedEvent.put("download_input_content_uri", event.get("download_input_content_uri"));
+            preparedEvent.put("download_output_content_uri", event.get("download_output_content_uri"));
+            preparedEvent.put("view_input_content_uri", event.get("view_input_content_uri"));
+            preparedEvent.put("view_output_content_uri", event.get("view_output_content_uri"));
+            try {
```

Comment:
because I get an unhandled exception that wasn't there before when dealing with just one event. with a list of events and `forEach()` the exception must be handled explicitly